### PR TITLE
feat(engine-v2/qp): Generate operation graph.

### PR DESCRIPTION
**Pull request by Finistere**
Simple initial PR to generate the operation graph with all fields and
requirements recursively. Taken from the tests:
```graphql
enum join__Graph {
  ACCOUNTS @join__graph(name: "accounts", url: "http://accounts:4001/graphql")
  INVENTORY @join__graph(name: "inventory", url: "http://inventory:4002/graphql")
  PRODUCTS @join__graph(name: "products", url: "http://products:4003/graphql")
  REVIEWS @join__graph(name: "reviews", url: "http://reviews:4004/graphql")
}

type Product
  @join__type(graph: INVENTORY, key: "upc")
  @join__type(graph: PRODUCTS, key: "upc")
  @join__type(graph: REVIEWS, key: "upc")
{
  upc: String!
  weight: Int @join__field(graph: INVENTORY, external: true) @join__field(graph: PRODUCTS)
  price: Int @join__field(graph: INVENTORY, external: true) @join__field(graph: PRODUCTS)
  inStock: Boolean @join__field(graph: INVENTORY)
  shippingEstimate: Int @join__field(graph: INVENTORY, requires: "price weight")
  name: String @join__field(graph: PRODUCTS)
  reviews: [Review] @join__field(graph: REVIEWS)
}

type Query
  @join__type(graph: ACCOUNTS)
  @join__type(graph: INVENTORY)
  @join__type(graph: PRODUCTS)
  @join__type(graph: REVIEWS)
{
  me: User @join__field(graph: ACCOUNTS)
  user(id: ID!): User @join__field(graph: ACCOUNTS)
  users: [User] @join__field(graph: ACCOUNTS)
  topProducts(first: Int = 5): [Product] @join__field(graph: PRODUCTS)
}

type Review
  @join__type(graph: REVIEWS, key: "id")
{
  id: ID!
  body: String
  product: Product
  author: User @join__field(graph: REVIEWS, provides: "username")
}

type User
  @join__type(graph: ACCOUNTS, key: "id")
  @join__type(graph: REVIEWS, key: "id")
{
  id: ID!
  name: String @join__field(graph: ACCOUNTS)
  username: String @join__field(graph: ACCOUNTS) @join__field(graph: REVIEWS, external: true)
  birthday: Int @join__field(graph: ACCOUNTS)
  reviews: [Review] @join__field(graph: REVIEWS)
}
```

with the following query

```graphql
    {
        topProducts {
            name
            reviews {
                author {
                    name
                }
            }
        }
    }
```

generates the following graph. The black part is the query itself, the blue part the resolvers and green requirements.

graphviz(7)

### CODE REVIEW

File: cli/crates/backend/src/api/publish.rs [hunk 1]
```diff
@@ -35,7 +35,22 @@ pub async fn publish(
         },
     });
 
-    let cynic::GraphQlResponse { data, errors } = client.post(api_url()).run_graphql(operation).await?;
+    let resp = client
+        .post(api_url())
+        .json(&operation)
+        .send()
+        .await
+        .unwrap()
```
Comment (yoav-lavi) [inline cli/crates/backend/src/api/publish.rs:43]: Did you mean to leave in these `unwrap`s and `println`?

Comment (Finistere) [reply cli/crates/backend/src/api/publish.rs:43]: wow hum no idea why it's in this PR, rebase didn't work as expected. good catch, thx

**Comment by tomhoule** [comment]
In the example graph in the description, why are there sometimes multiple `Field` edges between the same node? (for example topProducts -> upc on the left)

**Comment by Finistere** [comment]
> In the example graph in the description, why are there sometimes multiple `Field` edges between the same node? (for example topProducts -> upc on the left)

Yes that's an error, fixed it in another PR

**Comment by Finistere** [comment]
> difference between self.graph.operation and self.operation.

Same thing, just a `Deref`, I'm probably using them more than I should, don't know. They're certainly harder to read through GitHub for sure

**Review by tomhoule — APPROVED** [review]

**Pull request merged**